freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/cleanup/StructuredEventCleanupService.java

```diff
@@ -0,0 +1,46 @@
+package com.sequenceiq.freeipa.service.freeipa.cleanup;
+
+import static com.sequenceiq.cloudbreak.util.Benchmark.measureAndWarnIfLong;
+
+import java.util.Optional;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import com.sequenceiq.cloudbreak.structuredevent.repository.CDPStructuredEventRepository;
+import com.sequenceiq.cloudbreak.util.TimeUtil;
+import com.sequenceiq.freeipa.service.stack.StackService;
+
+@Service
+public class StructuredEventCleanupService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(StructuredEventCleanupService.class);
+
+    private static final int THREE_MONTHS = 3;
+
+    @Inject
+    private StackService stackService;
+
+    @Inject
+    private TimeUtil timeUtil;
+
+    @Inject
+    private CDPStructuredEventRepository structuredEventRepository;
+
+    public void cleanUpStructuredEvents(Long stackId) {
+        Optional<String> resourceCrn = stackService.findResourceCrnById(stackId);
+        if (resourceCrn.isPresent()) {
+            long timestampsThreeMonthsAgo = timeUtil.getTimestampThatMonthsBeforeNow(THREE_MONTHS);
+            LOGGER.debug("About to request deletion for FreeIPA structured events.");
+            String accountId = stackService.findAccountById(stackId).orElse(null);
+            measureAndWarnIfLong(() -> structuredEventRepository.deleteByAccountIdOlderThan(accountId, timestampsThreeMonthsAgo),
+                    LOGGER, "Cleaning up FreeIPA structured event(s) that are older than three months for the given account: " + accountId);
```

Comment:
I wouldn't call this with `null`. Check if accountid is present, and only call the `structuredEventRepository.deleteByAccountIdOlderThan` if it's present, instead of calling with `null`